.github/workflows/haskell-ci.yml

```diff
@@ -0,0 +1,69 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+jobs:
+  build-and-test:
+    services:
+      postgres:
+        image: postgres:15-alpine
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+          POSTGRES_USER: postgres
+          POSTGRES_DB: test
+        ports:
+          - 5432:5432
+        # Set health checks to wait until postgres has started
+        # needed because the postgres container does not provide a healthcheck
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    name: ${{ matrix.os }} / ghc ${{ matrix.ghc }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+        cabal:
+          - 3.6
+        ghc:
+          - 8.10.7
+          - 9.0.2
+          - 9.2.4
+
+    steps:
+      - uses: actions/checkout@v3
+        if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'
```

Comment:
Why are those conditions for this specific test? Were they actually meant as triggering events?

Reply:
TBH I don't have much idea what this means. I just copy pasted the config from elm-street and tweaked it. I think I'll just remove this line.